packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -119,16 +119,36 @@ class PostsRepo extends AbstractRepo<"Posts"> {
     return result?.meanKarma ?? 0;
   }
 
-  async getReadHistoryForUser(userId: string, limit: number): Promise<Array<DbPost & {lastUpdated: Date}>> {
+  async getReadHistoryForUser(
+    userId: string,
+    limit: number,
+    filter: { startDate?: Date, endDate?: Date, minKarma?: number, showEvents?: boolean } = {},
+    sort: {
+      karma?: boolean
+    } = {}
+  ): Promise<Array<DbPost & { lastUpdated: Date }>> {
+    const {startDate, endDate, minKarma, showEvents} = filter;
+
+    const startDateFilter = startDate ? `AND p."postedAt" >= '${startDate.toISOString()}' ` : '';
+    const endDateFilter = endDate ? `AND p."postedAt" <= '${endDate.toISOString()}' ` : '';
+    const baseScoreFilter = minKarma ? `AND p."baseScore" >= ${minKarma} ` : '';
+    const showEventsFilter = showEvents === false ? 'AND p."isEvent" IS NOT TRUE' : '';
+
```

Comment:
it seems plausible to me that it can be nice to reuse part of mongo > postgres pipeline for this (so users can still specify filters in the common mongo format 🤔